### PR TITLE
Fix memory leaks (#42)

### PR DESCRIPTION
**Pull request by r4f4**
Make sure to free the GStrv returned by modulemd.

Signed-off-by: Rafael Fonseca <r4f4rfs@gmail.com>

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -132,10 +132,10 @@ add_module_rpm_artifacts (Pool           *pool,
                           ModulemdModule *module,
                           Id              sdep)
 {
-  GStrv rpm_artifacts = modulemd_simpleset_dup (modulemd_module_peek_rpm_artifacts (module));
+  g_auto(GStrv) rpm_artifacts = modulemd_simpleset_dup (modulemd_module_peek_rpm_artifacts (module));
   g_auto(Queue) sel;
   queue_init (&sel);
-  for (; *rpm_artifacts; rpm_artifacts++)
+  for (GStrv artifacts = rpm_artifacts; *artifacts; artifacts++)
```
Comment (mmathesius) [inline main.c]: I'd suggest calling the loop variable `artifact` (singular).

File: main.c [hunk 2]
```diff
@@ -132,10 +132,10 @@ add_module_rpm_artifacts (Pool           *pool,
...
     {
       const char *nevra = *rpm_artifacts;
```
Comment (mmathesius) [inline main.c]: Should be `const char *nevra = *artifact;`

**Review by mmathesius — CHANGES_REQUESTED** [review]

**Comment by mmathesius** [comment]
Wow. The above changes (along the bug fix) completely zero the the valgrind "definitely" and "indirectly" lost bytes, leaving just 1.5k of "possibly" lost bytes!

File: main.c (hunk 2 above)
Comment (r4f4) [reply main.c]: Ops, tks. I tested with the correct fix but forgot to include it in the patch.

**Pull request merged**